python/spacewalk/satellite_tools/appstreams.py

```diff
@@ -0,0 +1,297 @@
+                for rpm in stream.get_rpm_artifacts():
+                    nevra = self._parse_rpm_name(rpm)
+
+                    log(2, "  Importing module packages.")
+                    pid = self._insert_module_package(nevra, module_id)
+                    if pid:
+                        log(3, f"{nevra} is pid={pid}")
+                        no_stream_added += 1
+                        no_total_added += 1
+                    else:
+                        log(
+                            2,
+                            f"{nevra} is not in the repository or is already imported.",
+                        )
+
+                log(
+                    0,
+                    f"{no_stream_added} new package(s) imported for module {stream.get_NSVCA()}.",
+                )
+
+        rhnSQL.commit()
+        log(2, f"{no_total_added} of {no_total_pkgs} packages matched.")
+
+    def _index_modulemd(self):
+        """
+        Indexes the Modulemd file.
+        """
+        idx = Modulemd.ModuleIndex.new()
+        idx.update_from_file(self.modulemd_file, True)
+        self.modulemd_index = idx
+
+    def _get_modules(self):
+        """
+        Retrieves modules from the Modulemd index.
+
+        Returns:
+        list: List of Modulemd.ModuleStreamV2 objects.
+        """
+        return [
+            self.modulemd_index.get_module(name)
+            for name in self.modulemd_index.get_module_names()
+        ]
+
+    def _insert_module(self, module: Nsvca):
+        """
+        Inserts a module into the database.
+
+        Parameters:
+        - module (Nsvca): Module to insert.
+
+        Returns:
+        int: ID of the inserted module.
+        """
+        q_insert_module = rhnSQL.prepare(
+            """
+            INSERT INTO suseAppstream (id, channel_id, name, stream, version, context, arch)
+            VALUES (sequence_nextval('suse_as_module_seq'), :chid, :n, :s, :v, :c, :a)
+            ON CONFLICT (name, stream, version, context, arch)
+            DO UPDATE SET name = :n, stream = :s, version = :v, context = :c, arch = :a
+            RETURNING id
+        """
+        )
+
+        q_insert_module.execute(
+            chid=self.channel_id,
+            n=module.name,
+            s=module.stream,
+            v=module.version,
+            c=module.context,
+            a=module.arch,
+        )
+
+        # Return the newly inserted module ID
+        row = q_insert_module.fetchone()
+        return row[0] if row else None
+
+    def _insert_module_apis(self, apis, module_id: int):
+        """
+        Inserts API entries for a module into the database.
+
+        Parameters:
+        - apis (list(str)): List of API entries to insert.
+        - module_id (int): ID of the module that the APIs belong to.
+        """
+        sql = "INSERT INTO suseAppstreamApi (rpm, module_id) VALUES "
+        sql += ",".join([f"('{api}', {module_id})" for api in apis])
+        sql += " ON CONFLICT DO NOTHING"
```

Comment:
Is this safe against SQL injections? I think we're handling untrusted input here.